windows/devices/uwb/UwbSession.cxx

```diff
@@ -14,12 +16,152 @@ UwbSession::UwbSession(std::weak_ptr<uwb::UwbSessionEventCallbacks> callbacks, w
     m_handleDriver(std::move(handleDriver))
 {}
 
+template <class T>
+std::unique_ptr<UWB_APP_CONFIG_PARAM>
+GetUWB_APP_CONFIG_PARAM(const T &value)
+{
+    auto pdata = std::make_unique<uint8_t>(FIELD_OFFSET(UWB_APP_CONFIG_PARAM, paramValue[sizeof T]));
+}
+
+template <class T>
+UWB_SET_APP_CONFIG_PARAMS
+GetUWB_SET_APP_CONFIG_PARAMS()
+{
+}
+
+/**
+ * @brief This function only works for integer types, aka uint8_t, uint16_t, uint32_t, uint64_t
+ *
+ * @param value
+ * @param size
+ * @param dst
+ * @return uint8_t*
+ */
+uint8_t *
+WriteBytes(uint64_t value, std::size_t size, uint8_t *dst)
+{
+    switch (size) {
+    case 1: {
+        dst[0] = (uint8_t)value;
+        return &dst[1];
+    }
+    case 2: {
+        auto asdf = (uint16_t *)dst;
+        asdf[0] = (uint16_t)value;
+        return (uint8_t *)(&asdf[1]);
+    }
+    case 4: {
+        auto asdf = (uint32_t *)dst;
+        asdf[0] = (uint32_t)value;
+        return (uint8_t *)(&asdf[1]);
+    }
+    case 8: {
+        auto asdf = (uint64_t *)dst;
+        asdf[0] = (uint64_t)value;
+        return (uint8_t *)(&asdf[1]);
+    }
+    default:
+        throw std::runtime_error("wrong number of bytes requested");
+    };
+}
+
+template <int N>
+uint8_t *
+WriteBytes(std::array<uint8_t, N> value, unsigned int size, uint8_t *dst)
+{
+    memcpy(dst, value.data(), size);
+    return dst + size;
+}
+
+/**
+ * @brief Maps the param to the length of the value. If the value is 0, means a variable length and special code to handle.
+ * Taken from the FiRa UCI spec Table 29:APP Configuration IDs.
+ * TODO put this somewhere more appropriate.
+ */
+std::unordered_map<UWB_APP_CONFIG_PARAM_TYPE, unsigned int> appConfigParamLengths{
```

Comment:
I'm not sure I understand why we need this map. The input type has its own size, so we'll always know what it is and how much data we have to write based on the DDI structure field(s) we're filling in.

Reply:
I don't know if sizeof(enum) is guaranteed to return the right size so I think it's still a good idea to have this map. Especially for those enums of weird sizes like 3 elements (does the internal representation of such an enum always map to a uint8_t?)

Reply:
It looks like this is no longer used, so the point is probably moot. In general, since `enum class` entities were being used, and all of them had fixed base-type, their underlying type is always as it is explicitly specified, so can be relied upon for a size calculation if desired.